wire.go

```diff
@@ -230,8 +250,39 @@ func (srv *Server) Close() error {
 		return nil
 	}
 
+	srv.logger.Info("initiating server shutdown")
 	srv.closing.Store(true)
 	close(srv.closer)
-	srv.wg.Wait()
+
+	// Wait for connections to finish with configurable timeout
+	done := make(chan struct{})
+	go func() {
+		defer close(done)
+		srv.connectionWG.Wait()
+	}()
+
+	timeout := srv.GracefulShutdownTimeout
+	if timeout <= 0 {
+		timeout = 1 * time.Second // Default fallback
+	}
```

Comment:
Did you consider pulling the default up to where GracefulShutdownTimeout is set? 

Also you can make it package private (lower case 'g').

Reply:
> Also you can make it package private (lower case 'g').

The idea was to make the GracefulShutdownTimeout configurable.